src/ServiceControl.Monitoring/QueueLength/TaggedLongValueOccurrenceHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace ServiceControl.Monitoring.QueueLength
+{
+    using System.Threading.Tasks;
+    using Infrastructure;
+    using Messaging;
+    using NServiceBus;
+
+    public class TaggedLongValueOccurrenceHandler : IHandleMessages<TaggedLongValueOccurrence>
+    {
+        readonly QueueLengthStore queueLengthStore;
+
+        public TaggedLongValueOccurrenceHandler(QueueLengthStore queueLengthStore)
+        {
+            this.queueLengthStore = queueLengthStore;
+        }
+
+        public Task Handle(TaggedLongValueOccurrence message, IMessageHandlerContext context)
+        {
+            var instanceId = EndpointInstanceId.From(context.MessageHeaders);
+            var messageType = context.MessageHeaders[MetricHeaders.MetricType];
+
+            if (message.Entries.Length == 0)
+            {
+                return TaskEx.Completed;
+            }
+
+            var value = message.Entries[0].Value;
+
+            switch (messageType)
+            {
+                case QueueLengthSent:
+                    queueLengthStore.UpdateSentSequence(message.TagValue, value);
+                    break;
+                case QueueLengthReceived:
+
+                    var virtualQueueId = new VirtualQueueId
+                    {
+                        EndpointName = instanceId.EndpointName,
+                        QueueName = WHAT,
```

Comment:
This is a breadcrumb I left. Could somebody help with figuring out how this should be set up? Also, answering this is required to make the PR reviewed properly reviewed (it's worth it only when we can obtain the queue name somehow). Otherwise it could require us to send two tags or sth similar https://github.com/Particular/ServiceControl.Monitoring.Data/pull/7

Reply:
@tmasternak Could you take a look at it?